src/LLMProviders/chatModelManager.ts

```diff
@@ -265,6 +265,7 @@ export default class ChatModelManager {
 
   async setChatModel(model: CustomModel): Promise<void> {
     const modelKey = getModelKeyFromModel(model);
+    setModelKey(modelKey);
```

Comment:
@logancyang Attention! This change is critical. Otherwise, there is a race condition that can lead to infinite loop. This setter must be called before the async getModelConfig call.